### PR TITLE
Use pytree from xla_client.

### PR DESCRIPTION
**Pull request by jblespiau**

### CODE REVIEW

File: jax/lib/__init__.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def _check_jaxlib_version():
 
 from jaxlib import xla_client
 from jaxlib import lapack
-from jaxlib import pytree
```
Comment (hawkinsp) [inline jax/lib/__init__.py:54]: Because the C++ and Python parts of JAX are released separately, you'll either need to make this change conditional on the jaxlib version, or make it another PR after a jaxlib release and minimum version bump. I would probably just make it conditional on the jaxlib version.

**Review by hawkinsp — CHANGES_REQUESTED** [review]
You need to update the WORKSPACE file to point to a newer TF tree hash so the jaxlib that gets built includes your TF tree changes.

File: jax/lib/__init__.py (hunk 1 above)
Comment (jblespiau) [reply jax/lib/__init__.py:54]: Trying this new approach.

File: build/install_xla_in_source_tree.sh [hunk 2]
```diff
@@ -52,8 +52,6 @@ fi
 
 # Copy the XLA dependencies into jax/lib, fixing up some imports to point to the
 # new location.
-cp -f "$(rlocation __main__/jaxlib/lapack.so)" "${TARGET}/jaxlib"
```
Comment (hawkinsp) [inline build/install_xla_in_source_tree.sh:55]: I think you didn't mean to drop the `lapack` line here?

Comment (jblespiau) [reply build/install_xla_in_source_tree.sh:55]: Thanks for the extra attention here. Sorry for the number of comments and attention you had to spend on this.

**Review by hawkinsp — APPROVED** [review]
